src/hooks/useNotificationSound.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react";
+import useSound from "use-sound";
+
+import notificationSound from "assets/sounds/newmessage.m4a";
+
+import { PrivateChatMessage, ChatTypes } from "types/chat";
+
+import { useChatSidebarControls } from "hooks/chatSidebar";
+
+import { useUser } from "./useUser";
+import { WithId } from "utils/id";
+
+export const useNotificationSound = (
+  privateChatMessages?: WithId<PrivateChatMessage>[]
+) => {
+  const { user } = useUser();
+  const userId = user?.uid;
+
+  const { chatSettings } = useChatSidebarControls();
+
+  const [play] = useSound(notificationSound, {
+    // `interrupt` ensures that if the sound starts again before it's
+    // ended, it will truncate it. Otherwise, the sound can overlap.
+    interrupt: true,
+  });
+
+  const [privateChatMessagesState, setPrivateChatMessagesState] = useState(
+    privateChatMessages
+  );
+
+  const chatWithUser =
+    chatSettings.openedChatType === ChatTypes.PRIVATE_CHAT
+      ? chatSettings.recipientId
+      : undefined;
+
+  useEffect(() => {
+    const newMessage = privateChatMessages?.filter(
+      ({ id: id1 }) =>
+        !privateChatMessagesState?.some(({ id: id2 }) => id2 === id1)
+    );
+
+    if (privateChatMessagesState === undefined) {
+      setPrivateChatMessagesState(privateChatMessages);
+    }
+
+    if (
+      privateChatMessages !== undefined &&
+      privateChatMessagesState !== undefined &&
+      newMessage !== undefined &&
+      newMessage.length !== 0 &&
+      privateChatMessages.length !== privateChatMessagesState.length &&
+      !newMessage[0].isRead &&
+      newMessage[0].from !== userId
+    ) {
```

Comment:
We're using TypeScript, so we can just use the `?.` 'optional chaining' navigator to reduce the verbosity here:

- https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

```suggestion
    if (
      privateChatMessages?.length !== privateChatMessagesState?.length &&
      newMessage?.length > 0
      !newMessage[0].isRead &&
      newMessage[0].from !== userId
    ) {
```

Reply:
[note] I last asked about the logic here in https://github.com/sparkletown/sparkle/pull/1262#discussion_r594125910 and it was explained in https://github.com/sparkletown/sparkle/pull/1262#discussion_r597416377

Reply:
[FYI] My suggested changes in https://github.com/sparkletown/sparkle/pull/1262#discussion_r604633362 likely override my comments/suggestions made above.